src/libcore/cmp.rs

```diff
@@ -512,16 +571,56 @@ impl PartialOrd for Ordering {
     }
 }
 
-/// Trait for values that can be compared for a sort-order.
+/// Trait for values that can be partially ordered.
+///
+/// This traits defines the operators `<`, `>`, `<=`, and `>=` which compare two values and return `bool`, 
+/// where `<` and `>` form [_strict_ partial ordering relations][wiki_porder].
+///
+/// The `PartialOrd` trait  has fewer requirements of the ordering relation than Ord. In particular, `partial_cmp` 
+/// can return `None` if two elements cannot be compared. That means that `PartialOrd` can be implemented 
+/// for types with incomparable values like floating-point numbers (`NaN`) and for ordering relations like 
+/// subset relations.
+///
+/// It extends the partial equivalence relation provided by `PartialEq` (`==`) with `partial_cmp(a, b) -> Option<Ordering>`, 
+/// which is a [trichotomy](https://en.wikipedia.org/wiki/Trichotomy_(mathematics)) of the ordering relation when 
+/// its result is `Some`:
+///
+/// * `a < b` if and only if `partial_cmp(a, b) == Some(Less)`
+/// * `a > b` if and only if `partial_cmp(a, b) == Some(Greater)`
+/// * `a == b` if and only if `partial_cmp(a, b) == Some(Equal)`
 ///
-/// The comparison must satisfy, for all `a`, `b` and `c`:
+/// and the absence of an ordering between `a` and `b` if and only if `partial_cmp(a, b) == None`. 
+/// Furthermore, this trait defines `<=` as `a < b || a == b` and `>=` as `a > b || a == b`.
 ///
-/// - antisymmetry: if `a < b` then `!(a > b)`, as well as `a > b` implying `!(a < b)`; and
-/// - transitivity: `a < b` and `b < c` implies `a < c`. The same must hold for both `==` and `>`.
+/// The comparisons must satisfy, for all `a`, `b`, and `c`:
 ///
-/// Note that these requirements mean that the trait itself must be implemented symmetrically and
-/// transitively: if `T: PartialOrd<U>` and `U: PartialOrd<V>` then `U: PartialOrd<T>` and `T:
-/// PartialOrd<V>`.
+/// * _transitivity_: if `a < b` and `b < c` then `a < c`
+/// * _duality_: if `a < b` then `b > a`
```

Comment:
Shouldn't it say somewhere that consistency with `==` from `PartialEq` is required?

Basically, this trait contains *strictly more* information than `PartialEq`, and it must be the case that `==` behaves exactly as if implemented via `partial_cmp == Some(Equal)`.